### PR TITLE
More copyediting

### PR DESCRIPTION
**Pull request by austinzheng**
@jpsim @tgoyne @bdash 

Changes:
- Copyedited RLMObject and RLMObjectSchema
- Changed certain repeated copy throughout the source to read the same

### CODE REVIEW

File: Realm/RLMObject.h [hunk 1]
```diff
@@ -49,15 +51,17 @@ NS_ASSUME_NONNULL_BEGIN
...
+ - `RLMArray<X>`, where `X` is an `RLMObject` subclass, so you can have many-to-many relationships.
 
  ### Querying
  
- You can query an object directly via the class methods: `allObjects`, `objectsWhere:`, and `objectsWithPredicate:`.
- These methods allow you to easily query a custom subclass for instances of this class in the
- default Realm. To search in a Realm other than the default Realm use the interface on an RLMRealm instance.
+ You can initiate queries directly via the class methods: `allObjects`, `objectsWhere:`, and `objectsWithPredicate:`.
+ These methods allow you to easily query a custom subclass for instances of that class in the
+ default Realm. To search in a Realm other than the default Realm, use the interface on an `RLMRealm` instance.
```
Comment (tgoyne) [inline Realm/RLMObject.h]: This last sentence has been wrong for a few years now (only the dynamic interface involves calling methods on RLMRealm instances for querying).

File: Realm/RLMObject.h [hunk 2]
```diff
@@ -67,18 +71,19 @@ NS_ASSUME_NONNULL_BEGIN
...
     KVC mutation methods on the containing class. You can simply call methods on
-    the RLMArray directly and the changes will be observed by the containing
+    the `RLMArray` directly; any changes will be automatically observed by the containing
     object.
- 2. Standalone `RLMObjects` cannot be added to a Realm while they have any
+ 2. Unmanaged `RLMObject` instances cannot be added to a Realm while they have any
     observed properties.
- 3. Modifying persisted `RLMObjects` in `-observeValueForKeyPath:ofObject:change:context:`
-    is problematic. Properties may change when the Realm is not in a write
+ 3. Modifying managed `RLMObject`s using `-observeValueForKeyPath:ofObject:change:context:`
```
Comment (bdash) [inline Realm/RLMObject.h]: I think "within" would make more sense than "using" here.

File: Realm/RLMObject.h [hunk 3]
```diff
@@ -122,194 +127,201 @@ NS_ASSUME_NONNULL_BEGIN
...
  
- @param realm   The Realm in which this object is persisted.
- @param value   The value used to populate the object. This can be any key/value coding compliant
-                object, or a JSON object such as those returned from the methods in NSJSONSerialization, or
-                an NSArray with one object for each persisted property. An exception will be
-                thrown if any required properties are not present and no default is set.
-                
-                When passing in an NSArray, all properties must be present, valid and in the same order as the properties defined in the model.
+ @param realm   The Realm which should manage the newly-created object.
+ @param value   A value used to populate the object. This can be any key/value coding compliant
```
Comment (bdash) [inline Realm/RLMObject.h]: Apple's documentation spells this "key-value coding".

File: Realm/RLMObject.h [hunk 4]
```diff
@@ -122,194 +127,201 @@ NS_ASSUME_NONNULL_BEGIN
...
+                object, or a JSON object such as those returned from the methods in `NSJSONSerialization`, or
```
Comment (bdash) [inline Realm/RLMObject.h]: It feels strange to me to refer to "a JSON object" here. The objects returned from `NSJSONSerialization` are just plain ol' `NSDictionary`s and `NSArray`s. There's nothing JSONy about them.

File: Realm/RLMObject.h [hunk 5]
```diff
@@ -122,194 +127,201 @@ NS_ASSUME_NONNULL_BEGIN
...
+ This method can only be called on Realm object types with a primary key defined. If there is already
+ an object with the same primary key value in the default Realm, its values are updated and the object
+ is returned. Otherwise, this method creates and populates a new instance of the object in the default Realm.
  
  If nested objects are included in the argument, `createOrUpdateInDefaultRealmWithValue:` will be
- called on them if have a primary key (`createInDefaultRealmWithValue:` otherwise).
+ recursively called on them if have a primary key, `createInDefaultRealmWithValue:` if they do not.
  
- This is a no-op if the argument is an RLMObject of the same type already backed by the target realm.
+ If there is a Realm object managed by the default
```
Comment (bdash) [inline Realm/RLMObject.h]: Partial sentence?

Comment (austinzheng) [inline Realm/RLMObject.h]: Yes. Not sure how that ended up there.

File: Realm/RLMObject.h [hunk 6]
```diff
@@ -122,194 +127,201 @@ NS_ASSUME_NONNULL_BEGIN
...
+
+ If the argument is a Realm object already managed by the default Realm, the argument's type is the same
```
Comment (austinzheng) [inline Realm/RLMObject.h]: I would appreciate any opinions over how this sentence is worded. I think it still might not be entirely clear and/or accurate.

File: Realm/RLMObject.h (hunk 4 above)
Comment (austinzheng) [inline Realm/RLMObject.h]: Would something like "or an array or dictionary returned from the methods in `NSJSONSerialization`" be okay in this context? (This same overall description shows up multiple times; I'll update them all.)

File: Realm/RLMObject.h (hunk 1 above)
Comment (austinzheng) [inline Realm/RLMObject.h]: Excellent. I'll fix this.

**Comment by austinzheng** [comment]
I updated based on feedback. Also fixed more shared copy.

**Comment by jpsim** [comment]
FYI two small SwiftLint warnings: https://travis-ci.org/realm/realm-cocoa/jobs/132426310

```
Object.swift:93: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
Realm.swift:289: warning: Line Length Violation: Line should be 120 characters or less: currently 121 characters (line_length)
```

**Comment by austinzheng** [comment]
Thanks @jpsim! SwiftLint is super useful.

**Comment by austinzheng** [comment]
If there are no further comments I'll merge this as soon as the most recent CI build passes.

**Pull request merged**